tests/spec/s-xregexp.js

```diff
@@ -182,6 +182,19 @@ describe('XRegExp()', function() {
         expect(function() {XRegExp('(\\(?:)');}).not.toThrow();
     });
 
+    it('should not put (?:) at beginning/end of groups', function() {
+        var regex = XRegExp('( [0-9]{4} ) -?  # year  \n' +
+                            '( [0-9]{2} ) -?  # month \n' +
+                            '( [0-9]{2} )     # day     ', 'x');
+        expect(regex.source).toEqual('([0-9]{4})(?:)-?(?:)([0-9]{2})(?:)-?(?:)([0-9]{2})(?:)');
```

Comment:
Oops, yeah it is a bit brittle. We could change it to reject certain substrings, but then we might end up duplicating some of the logic in the non-test code. What if we changed it to something a little simpler, but still future-proof, like this?

```js
expect(regex.source.length <= 54).toBe(true); // 54 is the length of the current result
```